### PR TITLE
[34]: add accessor method for some class members

### PR DESCRIPTION
**Pull request by skaupper**
extend builder tests to check for these getters

refactor Keys.h and Color.h
update comparison operators for Geometry.h and Color.h

### CODE REVIEW

File: src/utils/Geometry.cpp [hunk 1]
```diff
@@ -21,8 +39,38 @@ std::string Point::toString() const
...
+    } else if (cmp == -1) {
+        return false;
+    }
+
+    return false;
+}
+
 
-Size::Size() : w(0), h(0)
+Size::Size() : w(100), h(100)
```
Comment (skaupper) [inline src/utils/Geometry.cpp:73]: width and height are relative: default is 100% of parent

File: src/utils/Geometry.cpp [hunk 2]
```diff
@@ -3,6 +3,24 @@
 using namespace bkengine;
 
 
+static double EPSILON = 0.0001;
+
+static short doubleCompare(double d1, double d2)
```
Comment (chrboe) [inline src/utils/Geometry.cpp]: uint8_t

File: src/utils/Geometry.cpp [hunk 3]
```diff
@@ -40,8 +88,38 @@ std::string Size::toString() const
...
+}
+
+bool Size::operator!=(const Size &size) const
+{
+    return !(operator==(size));
+}
+
+bool Size::operator<(const Size &size) const
+{
+    short cmp;
```
Comment (chrboe) [inline src/utils/Geometry.cpp]: uint8_t

File: src/utils/Geometry.cpp [hunk 4]
```diff
@@ -71,15 +149,44 @@ std::string Rect::toString() const
...
 bool Rect::operator!=(const Rect &r) const
 {
-    return !(r == *this);
+    return !(operator==(r));
 }
 
 bool Rect::operator<(const Rect &r) const
 {
-    return r != *this;
+    short cmp;
```
Comment (chrboe) [inline src/utils/Geometry.cpp]: uint8_t

**Review by chrboe — CHANGES_REQUESTED** [review]

**Review by chrboe — APPROVED** [review]

**Pull request merged**